src/lib/api/prefetch.ts

```diff
@@ -0,0 +1,200 @@
+  permissions?: PrefetchOptionsPermissions;
+  updated_config?: {
+    fromSources?: string[];
+  };
+};
+
+// ==>
+// {
+//   "apikey": "AHvhedybhQMqZOqRvZquez",
+//   security: {
+//     policy: "",
+//     signature: ""
+//   },
+// 	"permissions": ["transforms_ui", "gmail"], -- events
+// 	"settings": ["inapp_browser", "customsource"],
+// 	"events": ["picker"], -- events
+// 	"picker_config": {
+// 		"fromSources": ["googledrive", "dropbox"]
+// 	}
+// }
+
+// <==
+// {
+//   "blocked": false,
+//   "settings": {
+//     "customsource": false,
+//     "inapp_browser": true
+//   },
+//   "permissions": {
+//     "transforms_ui": false
+//   },
+//   "updated_config": {
+//     "fromSources": [
+//       "googledrive"
+//     ]
+//   }
+// }
+
+/**
+ * @private
+ */
+export class Prefetch {
+  private session: Session;
+
+  private prefetchUrl: string;
+
+  private configToCheck: PickerOptions;
+
+  constructor(session: Session) {
+    this.session = session;
+    this.prefetchUrl = session.urls.cloudApiUrl;
+  }
+
+  async getConfig({ pickerOptions, settings, permissions, events }: PrefetchOptions) {
+    if (this.session.prefetch) {
+      await requestWithSource().post(`${this.prefetchUrl}/prefetch`, {
+        events,
+      });
+
+      return Promise.resolve(this.session.prefetch);
+    }
+
+    const configToSend = this.cleanUpCallback(pickerOptions);
+
+    let paramsToSend: SendObject = {
+      apikey: this.session.apikey,
+      permissions: Object.keys(permissions),
+      settings: Object.keys(settings),
+      pickerOptions: configToSend,
+      events,
+    };
+
+    if (this.session.policy) {
+      Object.assign(paramsToSend, { security: { policy: this.session.policy } });
+    }
+
+    if (this.session.signature) {
+      Object.assign(paramsToSend, { security: { signature: this.session.signature } });
+    }
+
+    const response = await requestWithSource()
+      .post(`${this.prefetchUrl}/prefetch`, paramsToSend)
+      .then(res => res.data)
+      .then(data => data);
+
+    return this.reassignCallbacks(response);
+  }
+
+  private cleanUpCallback(pickerOptions: PickerOptions) {
+    this.configToCheck = cloneDeep.clone(pickerOptions);
+
+    const pickerDropPaneKey = ['onDragEnter', 'onDragLeave', 'onDragOver', 'onDrop', 'onSuccess', 'onError', 'onProgress', 'onClick'];
+
+    const pickerKey = [
+      'onClose',
+      'onOpen',
+      'onFileSelected',
+      'onFileUploadStarted',
+      'onFileUploadFinished',
+      'onFileUploadFailed',
+      'onFileUploadProgress',
+      'onFileCropped',
+      'onUploadStarted',
+      'onUploadDone',
+    ];
+
+    Object.keys(this.configToCheck).map(key => {
+      const keyName = this.configToCheck[key];
+      if (pickerKey.includes(keyName) || pickerDropPaneKey.includes(keyName)) {
+        this.configToCheck[key] = null;
+      }
+    });
+
+    return this.configToCheck;
+  }
+
+  private reassignCallbacks(response: responseObject) {
+    let reassignResponse = {};
```

Comment:
In this function, you have to overwrite old copy of the config this.configToCheck with new one incoming from the response. You can use  one of those functions https://stackoverflow.com/questions/19965844/lodash-difference-between-extend-assign-and-merge

so as the result we will have picker config with new values from response and all callbacks defined previously by user

Reply:
done